webapi.py

```diff
@@ -0,0 +1,127 @@
+import itertools
+import json
+import uuid
+
+from flask import Blueprint, jsonify, request
+
+from models import Shred, Cluster, ClusterMember
+
+
+app = Blueprint("webapi", __name__)
+
+
+@app.route('/cluster', methods=["GET"])
+@app.route('/cluster/<cluster_id>', methods=["GET"])
+def get_cluster(cluster_id=None):
+    """Renders a single cluster as json.
+
+    If the cluster id is not provided, picks a random one.
+    """
+    batch = request.args.get("batch")
+    if cluster_id is None:
+        cluster = Cluster.get_some(batch=batch)
+    else:
+        cluster = Cluster.objects.get_or_404(pk=cluster_id)
+
+    return jsonify({
+        "success": True,
+        "data": {
+            "cluster": json.loads(cluster.to_json()),
+        },
+    })
+
+
+@app.route('/shred/<shred_id>')
+def get_shred(shred_id):
+    """Looks up a single shred by id."""
+    shred = Shred.objects.get_or_404(id=shred_id)
+    return jsonify({'success': True,
+                    'data': {
+                        'shred': json.loads(shred.to_json())
+                    }})
+
+
+@app.route('/cluster', methods=['POST'])
+def create_cluster():
+    """Merges two clusters.
+
+    Processes a POST request that should contain an object like:
+    { "cluster": {
+        "parents": ["parent1_id", "parent2_id"],
+        "members": [{
+            "shred": "shred1_id",
+            "position": [100, 500],
+            "angle": 35,
+          },
+          ...
+        ]
+    } }
+
+    """
+
+    req = request.get_json().get('cluster')
+
+    parents = map(lambda pk: Cluster.objects.get_or_404(pk=pk), req['parents'])
```

Comment:
No :(
InvalidQueryError: Not a query object: fixtures1:6_3. Did you intend to use key=value?